docs/site/Repository-generator.md

```diff
@@ -0,0 +1,97 @@
+---
+lang: en
+title: 'Repository generator'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Repository-generator.html
+---
+
+{% include content/generator-create-app.html lang=page.lang %}
+
+### Synopsis
+
+Adds a new [Repository or Multiple Repositories](Repositories.md) class to a
+LoopBack application with one single command.
+
+### Notes
+
+The optional argument `[<name>]`, referes to a valid datasource already created
```

Comment:
> Either way, I think the datasource should be specified via --dataSource argument?

I agree. 

> I find it confusing that names is referring to datasource in lb4 repository command.

It is extending from ArtifactGenerator and because of the multiple selection of models and thus mulitple repository generation feature:

1. I couldn't see the name as the repository name because it can generate multiple ones
2. ArtifactGenerator still was showing the name in the help see `lb4 controller --help` for instance 
3. I agree that it should be datasource name and this can be discarded
4. I will be placeing the option --dataSource instead

Reply:
Maybe your generator should be inheriting from a different base generator class?

Reply:
probably the option is BaseGenerator, I just followed up the other options initially. let me change it and try it.